includes/type/object/class-collection-stats-type.php

```diff
@@ -0,0 +1,380 @@
+								0 < $per_page ? $per_page : null
+							);
+				
+							return array_map(
+								static function ( $name, $terms ) {
+									return (object) compact( 'name', 'terms' );
+								},
+								array_keys( $attribute_counts ),
+								array_values( $attribute_counts )
+							);
+						},
+					],
+					'ratingCounts'      => [
+						'type'        => [ 'list_of' => 'RatingCount' ],
+						'args'        => [
+							'page'    => [
+								'type'        => 'Int',
+								'description' => __( 'Page of results to return', 'wp-graphql-woocommerce' ),
+							],
+							'perPage' => [
+								'type'        => 'Int',
+								'description' => __( 'Number of results to return per page', 'wp-graphql-woocommerce' ),
+							],
+						],
+						'description' => __( 'Returns number of products with each average rating', 'wp-graphql-woocommerce' ),
+						'resolve'     => static function ( $source, $args ) {
+							$page          = ! empty( $args['page'] ) ? $args['page'] : 1;
+							$per_page      = ! empty( $args['perPage'] ) ? $args['perPage'] : 0;
+							$rating_counts = ! empty( $source['rating_counts'] ) ? $source['rating_counts'] : [];
+							$rating_counts = array_slice(
+								$rating_counts,
+								( $page - 1 ) * $per_page,
+								0 < $per_page ? $per_page : null
+							);
+				
+							return $rating_counts;
+						},
+					],
+					'stockStatusCounts' => [
+						'type'        => [ 'list_of' => 'StockStatusCount' ],
+						'args'        => [
+							'page'    => [
+								'type'        => 'Int',
+								'description' => __( 'Page of results to return', 'wp-graphql-woocommerce' ),
+							],
+							'perPage' => [
+								'type'        => 'Int',
+								'description' => __( 'Number of results to return per page', 'wp-graphql-woocommerce' ),
+							],
+						],
+						'description' => __( 'Returns number of products with each stock status', 'wp-graphql-woocommerce' ),
+						'resolve'     => static function ( $source, $args ) {
+							$page                = ! empty( $args['page'] ) ? $args['page'] : 1;
+							$per_page            = ! empty( $args['perPage'] ) ? $args['perPage'] : 0;
+							$stock_status_counts = ! empty( $source['stock_status_counts'] ) ? $source['stock_status_counts'] : [];
+							$stock_status_counts = array_slice(
+								$stock_status_counts,
+								( $page - 1 ) * $per_page,
+								0 < $per_page ? $per_page : null
+							);
+				
+							return $stock_status_counts;
+						},
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Prepare the WP_Rest_Request instance used for the resolution of a
+	 * statistics for a product connection.
+	 * 
+	 * @param array $where_args  Arguments used to filter the connection results.
+	 * 
+	 * @return \WP_REST_Request
+	 */
+	public static function prepare_rest_request( array $where_args = [] ) /* @phpstan-ignore-line */ {  
+		$request = new \WP_REST_Request();
+		if ( empty( $where_args ) ) {
+			return $request;
+		}
+
+		$key_mapping = [
```

Comment:
Please add a filter here

Reply:
Not at this time, because the functionality of ProductQueryFilters is kinda limited at the moment and not extendable by hooks or extra params, we can revisit this once it has been expanded upon or made extendable is some way 🤔.